src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py

```diff
@@ -946,52 +946,314 @@ def test_network_app_gateway_waf_v2_config(self, resource_group):
 
 class NetworkAppGatewayWafPolicyScenarioTest(ScenarioTest):
 
-    @ResourceGroupPreparer(name_prefix='cli_test_app_gateway_waf_policy')
-    def test_network_app_gateway_waf_policy(self, resource_group):
+    @ResourceGroupPreparer(name_prefix='cli_test_app_gateway_waf_policy_')
+    def test_network_app_gateway_waf_policy_with_application_gateway(self, resource_group):
+        self.kwargs.update({
+            'waf': 'agp1',
+            'custom-rule1': 'rule1',
+            'custom-rule2': 'rule2',
+            'ip1': 'pip1',
+            'ag1': 'ag1',
+            'ip2': 'pip2',
+            'ag2': 'ag2',
+            'rg': resource_group,
+            'csr_grp1': 'REQUEST-921-PROTOCOL-ATTACK',
+            'csr_grp2': 'REQUEST-913-SCANNER-DETECTION'
+        })
 
-        self.kwargs.update({
-            'wafp': 'agp1',
+        # create a waf policy
+        self.cmd('network application-gateway waf-policy create -g {rg} -n {waf}')
+        self.cmd('network application-gateway waf-policy update -g {rg} -n {waf} --tags test=best',
+                 checks=self.check('tags.test', 'best'))
+        self.cmd('network application-gateway waf-policy show -g {rg} -n {waf}')
+        self.cmd('network application-gateway waf-policy list -g {rg}',
+                 checks=self.check('length(@)', 1))
+
+        # add two custom rules of this waf-policy
+        self.cmd('network application-gateway waf-policy custom-rule create -g {rg} '
+                 '--policy-name {waf} -n {custom-rule1} '
+                 '--priority 50 --action log --rule-type MatchRule',
+                 checks=self.check('priority', 50))
+        self.cmd('network application-gateway waf-policy custom-rule create -g {rg} '
+                 '--policy-name {waf} -n {custom-rule2} '
+                 '--priority 100 --action log --rule-type MatchRule')
+
+        # update some policy settings of this waf-policy
+        self.cmd('network application-gateway waf-policy policy-setting update -g {rg} --policy-name {waf} '
+                 '--state Enabled --file-upload-limit-in-mb 64 --mode Prevention')
+
+        # add two managed rule set to the managed rules of this waf-policy
+        self.cmd('network application-gateway waf-policy managed-rule rule-set add -g {rg} --policy-name {waf} '
+                 '--type OWASP --version 3.0 '
+                 '--group-name {csr_grp1} --rules 921100 921110')
```

Comment:
How can user know 92110 921110?

Reply:
No. user can not. May be we should provide another like `list-managed-rule-set` to show all of this thing?

Reply:
list-managed-rule? How do you know these values? It's better to provide a command. At least, we can provide a link.

Reply:
`list-managed-rule-set` There is no such a link to list all managed-rule-set. I gave a link https://docs.microsoft.com/en-us/azure/web-application-firewall/afds/afds-overview But it doesn't help a lot I think.

There is a command `az network application-gateway waf-config list-rule-sets` doing something similar.

For now. user have to go to portal to find out which rule-set and rules he/she want's to disable.
By default, all of rules in a rule-set are Enabled.

Reply:
I found a link https://docs.microsoft.com/en-us/azure/web-application-firewall/ag/application-gateway-crs-rulegroups-rules for user to query rule sets and added it to waf-policy managed-rule rule-set add/update.

We should better need a list command to show all these rules in the feature.